app/connectors/connectors/http4/src/main/java/io/syndesis/connector/http4/Http4ConnectorFactory.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.connector.http4;
+
+import java.util.Map;
+
+import org.apache.camel.Endpoint;
+import org.apache.camel.component.extension.ComponentExtension;
+
+import io.syndesis.integration.component.proxy.ComponentDefinition;
+import io.syndesis.integration.component.proxy.ComponentProxyComponent;
+import io.syndesis.integration.component.proxy.ComponentProxyFactory;
+
+public class Http4ConnectorFactory implements ComponentProxyFactory {
+
+    @Override
+    public ComponentProxyComponent newInstance(String componentId, String componentScheme) {
+        return new Http4ProxyComponent(componentId, componentScheme);
+    }
+
+    private static class Http4ProxyComponent extends ComponentProxyComponent {
+
+        private String httpUri;
+
+        public Http4ProxyComponent(String componentId, String componentScheme) {
+            super(componentId, componentScheme);
+            registerExtension(this::getComponentVerifier);
+        }
+
+        private ComponentExtension getComponentVerifier() {
+            return new Http4ComponentVerifierExtension(getComponentId(), getCamelContext());
+        }
+
+        @Override
+        public void setOptions(Map<String, Object> options) {
+            // connection parameters
+            httpUri = (String) options.remove("httpUri");
```

Comment:
It is not clear to me how this should work as in the connector json, there's no reference to `httpUri` but there is as example `hostname` so I guess there should be some bits to build the uri from the parameters

Reply:
@lburgazzoli weirdly the http4.json was not included in the PR branch. I had taken a copy of the repo before the rebase so I've repushed this.  I switch over to the a single endpoint url instead of seperate host/port/ssl parameters to simplify the UI. Camel http4 does use this single, httpUri, parameter in it's format, but that can't be mapped through if we want to easily support ssl.

Reply:
@kcbabo is it ok to go with a single `httpUri` option or we should split it into different parts ?